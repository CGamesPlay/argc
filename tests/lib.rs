use insta::assert_snapshot;

#[macro_export]
macro_rules! assert_argc {
    (
        $source:expr,
        $args:expr
    ) => {
        let (stdout, stderr) = argc::run($source, $args).unwrap();
        assert_snapshot!(stderr.unwrap_or_default());
        assert_snapshot!(stdout.unwrap_or_default());
    };
}

#[test]
fn test_git() {
    assert_argc!(include_str!("git.sh"), &["git", "-h"]);
}

#[test]
fn test_git_add() {
    assert_argc!(include_str!("git.sh"), &["git", "add", "-h"]);
}

#[test]
fn test_git_remote() {
    assert_argc!(include_str!("git.sh"), &["git", "push", "-h"]);
}

#[test]
fn test_git_log() {
    assert_argc!(include_str!("git.sh"), &["git", "log", "-h"]);
}
