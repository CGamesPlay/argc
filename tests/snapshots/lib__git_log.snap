---
source: tests/lib.rs
assertion_line: 32
expression: stderr.unwrap_or_default()

---
git-log 
Shows the commit log.

USAGE:
    git log [OPTIONS]

OPTIONS:
        --decorate <DECORATE>    If no --decorate-refs is given, pretend as if all refs were
                                 included. [default: short] [possible values: short, full, auto, no]
        --follow                 Continue listing the history of a file beyond renames
        --grep <GREP>...         Limit the commits output to ones with log message that matches the
                                 specified pattern
    -h, --help                   Print help information

